### PR TITLE
Added the other radio protocols to the additional ports on RE1

### PR DESCRIPTION
**Pull request by yamahabest**
And added the ports to the getInputType function.

### CODE REVIEW

File: shared/uavobjectdefinition/hwbrainre1.xml [hunk 1]
```diff
@@ -55,8 +62,16 @@
 				<option>LighttelemetryTx</option>
 				<option>MavLinkTX</option>
 				<option>MavLinkTX_GPS_RX</option>
+				<option>VTX Config TBS SmartAudio</option>
```
Comment (yamahabest) [inline shared/uavobjectdefinition/hwbrainre1.xml:65]: Was also on MultiPortSerial2, figured it should/could be here also.

File: shared/uavobjectdefinition/hwbrainre1.xml [hunk 2]
```diff
@@ -55,8 +62,16 @@
...
 				<option>FrSKY Sensor Hub</option>
 				<option>FrSKY SPort Telemetry</option>
+				<option>S.Bus</option>
+				<option>S.Bus Non Inverted</option>
+				<option>DSM</option>
```
Comment (yamahabest) [inline shared/uavobjectdefinition/hwbrainre1.xml:70]: DSM should also be possible on these ports, right?

File: ground/gcs/src/plugins/boards_brainfpv/brainre1.cpp [hunk 3]
```diff
@@ -200,6 +200,72 @@ Core::IBoardType::InputType BrainRE1::getInputType()
...
+        return INPUT_TYPE_IBUS;
+    case HwBrainRE1::MULTIPORTSERIAL_SRXL:
+        return INPUT_TYPE_SRXL;
+    case HwBrainRE1::MULTIPORTSERIAL_TBSCROSSFIRE:
+        return INPUT_TYPE_TBSCROSSFIRE;
+    default:
+        break;
+    }
+    
+    if (settings.MultiPortMode == HwBrainRE1::MULTIPORTMODE_DUALSERIAL4PWM)
```
Comment (yamahabest) [inline ground/gcs/src/plugins/boards_brainfpv/brainre1.cpp:245]: Not sure if this check is needed?
Where can I see what's the default value of settings.MultiPortSerial2, if the MultiPortMode is set to "Normal"?

**Comment by mluessi** [comment]
What is the use case for this? The only reason I can see why someone would want to run the receiver on a different port is that the RxPort is damaged or for a Rx protocol that includes telemetry and needs the Tx pin (e.g. TBS Crossfire). Other than that, it will just confuse users and lead to problems, e.g., the GCS gives them the option to connect S.Bus to the serial port but it doesn't have an inverter..

**Comment by yamahabest** [comment]
Ok, ok, figured that it would also be strange to only have Crossfire on there.
No need to merge.

Would propose to keep the additional switch-cases with the Crossfire option from this change set, though.

**Comment by yamahabest** [comment]
If purging the addition of the protocols, would the "VTX Config TBS SmartAudio" addition on MultiPortSerial still be ok?

**Comment by mluessi** [comment]
It should work on MultiPortSerial2; it doesn't on MultiPortSerial due to the non-standard signal levels it uses which are not compatible with the bi-directional inverter on that port.

**Comment by yamahabest** [comment]
Ok, created a new PR ( https://github.com/d-ronin/dRonin/pull/1734 ) for just the addition of crossfire to the getInputType function.

**Comment by yamahabest** [comment]
https://github.com/d-ronin/dRonin/pull/1734

**Pull request closed without merging**